packages/features/tasker/internal-tasker.ts

```diff
@@ -0,0 +1,38 @@
+import { Task } from "./repository";
+import { type Tasker, type TaskTypes } from "./tasker";
+import tasksMap from "./tasks";
+
+/**
+ * This is the default internal Tasker that uses the Task repository to create tasks.
+ * It doens't have any external dependencies and is suitable for most use cases.
+ * To use a different Tasker, you can create a new class that implements the Tasker interface.
+ * Then, you can use the TaskerFactory to select the new Tasker.
+ */
+export class InternalTasker implements Tasker {
+  async create(type: TaskTypes, payload: string): Promise<void> {
+    await Task.create(type, payload);
+  }
+  async processQueue(): Promise<void> {
+    const tasks = await Task.getNextBatch();
+    const tasksPromises = tasks.map(async (task) => {
+      const taskHandlerGetter = tasksMap[task.type as keyof typeof tasksMap];
```

Comment:
If `task.type` is inferred from the TaskTypes, this cast should not be required, no?

Reply:
task type is of type of string since it comes directly from db. Didn't want to make it a enum to avoid adding more migrations when adding more tasks.